cfg/validate.go

```diff
@@ -107,6 +107,40 @@ func isValidKernelListCacheTTL(TTLSecs int64) error {
 	return nil
 }
 
+func isValidMetadataCache(c MetadataCacheConfig) error {
+	// Validate ttl-secs.
+	if c.TtlSecs != TtlInSecsUnsetSentinel {
```

Comment:
What if someone sets the ttl-secs to TtlInSecsUnsetSentinel via Viper. Validation won't fail here. And it will skip the rationalization as well since the concerned Viper key is set.
So, the value of ttl-secs will be set to TtlInSecsUnsetSentinel itself. While this case is improbable, I was wondering if we can handle it better.

Reply:
Made the changes to use IsSet implementation here as well instead of relying on UnsetSentinel values.